ledger/voters.go

```diff
@@ -224,17 +261,22 @@ func (vt *votersTracker) removeOldVoters(hdr bookkeeping.BlockHeader) {
 // not need any blocks, it returns base.
 func (vt *votersTracker) lowestRound(base basics.Round) basics.Round {
 	minRound := base
+
+	vt.votersMu.RLock()
+	defer vt.votersMu.RUnlock()
+
 	for r := range vt.votersForRoundCache {
 		if r < minRound {
 			minRound = r
 		}
 	}
+
 	return minRound
 }
 
-// getVoters() returns the top online participants from round r.
-func (vt *votersTracker) getVoters(r basics.Round) (*ledgercore.VotersForRound, error) {
-	tr, ok := vt.votersForRoundCache[r]
+// VotersForStateProof returns the top online participants from round r.
+func (vt *votersTracker) VotersForStateProof(r basics.Round) (*ledgercore.VotersForRound, error) {
+	tr, ok := vt.getVoters(r)
```

Comment:
Q: how about moving this into state proof? the voters state is only depends on top accounts from the ledger.
In this case we the state proof machinery only receives top accounts data and could process/save them asynchronously without blocking on tree's cond variable? It could simplify the code a lot.

Reply:
I'm not sure I follow. Which parts would you move into the stateproof package?